packages/terra-consumer-nav/src/components/popup/Popup.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import TerraPopup from 'terra-popup';
+import styles from './Popup.scss';
+
+const cx = classNames.bind(styles);
+
+
+const propTypes = {
+  /**
+   * Title to be rendered on top of the popup content
+   */
+  title: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.element,
+  ]),
+  /**
+   * Boolean value to hide/display popup header with title and close button
+   */
+  hasHeader: PropTypes.bool,
+  /**
+   * Content to be displayed in a dialog
+   */
+  popupContent: PropTypes.element,
+  /**
+   * Boolean value to render the popup
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * Required callback function for use by parent component to update state.
+   */
+  closePopup: PropTypes.func,
+};
+
+const defaultProps = {
+  title: null,
+  hasHeader: false,
+  popupContent: null,
+  isOpen: false,
+  closePopup: null,
+};
+
+const Popup = ({
+ title,
+ hasHeader,
+ popupContent,
+ isOpen,
+ closePopup,
+ ...customProps
+}) => {
+  const popupHeader = (
```

Comment:
since you have a prop for conditionally displaying the header, you can make this a lazy evaluated const to not run all this when the condition is false